staging/operator-lifecycle-manager/pkg/controller/operators/olm/plugins/downstream_csv_namespace_labeler_plugin.go

```diff
@@ -0,0 +1,368 @@
+// upstream Pod Security Admission controller, which will become active in k8s 1.25.
+// see https://kubernetes.io/docs/concepts/security/pod-security-admission/
+// If a CSV is created or modified, this controller will look at the csv's namespace. If it is a non-payload namespace,
+// if the namespace name is prefixed with 'openshift-', and if the namespace does not contain the label (whatever
+// value it may be set to), it will add the "security.openshift.io/scc.podSecurityLabelSync=true" to the namespace.
+type csvNamespaceLabelerPlugin struct {
+	namespaceLister       listerv1.NamespaceLister
+	nonCopiedCsvListerMap map[string]listerv1alpha1.ClusterServiceVersionLister
+	kubeClient            operatorclient.ClientInterface
+	externalClient        versioned.Interface
+	logger                *logrus.Logger
+}
+
+func NewCsvNamespaceLabelerPluginFunc(ctx context.Context, config OperatorConfig, hostOperator HostOperator) (OperatorPlugin, error) {
+
+	if hostOperator == nil {
+		return nil, fmt.Errorf("cannot initialize plugin: operator undefined")
+	}
+
+	plugin := &csvNamespaceLabelerPlugin{
+		kubeClient:            config.OperatorClient(),
+		externalClient:        config.ExternalClient(),
+		logger:                config.Logger(),
+		namespaceLister:       nil,
+		nonCopiedCsvListerMap: map[string]listerv1alpha1.ClusterServiceVersionLister{},
+	}
+
+	plugin.log("setting up csv namespace plug-in for namespaces: %s", config.WatchedNamespaces())
+
+	namespaceInformer := newNamespaceInformer(config.OperatorClient(), config.ResyncPeriod()())
+
+	plugin.log("registering namespace informer")
+
+	plugin.namespaceLister = listerv1.NewNamespaceLister(namespaceInformer.GetIndexer())
+
+	namespaceQueue := workqueue.NewNamedRateLimitingQueue(
+		workqueue.DefaultControllerRateLimiter(),
+		"csv-ns-labeler-plugin-ns-queue",
+	)
+	namespaceQueueInformer, err := queueinformer.NewQueueInformer(
+		ctx,
+		queueinformer.WithInformer(namespaceInformer),
+		queueinformer.WithLogger(config.Logger()),
+		queueinformer.WithQueue(namespaceQueue),
+		queueinformer.WithIndexer(namespaceInformer.GetIndexer()),
+		queueinformer.WithSyncer(plugin),
+	)
+	if err != nil {
+		return nil, err
+	}
+	if err := hostOperator.RegisterQueueInformer(namespaceQueueInformer); err != nil {
+		return nil, err
+	}
+
+	for _, namespace := range config.WatchedNamespaces() {
+		plugin.log("setting up namespace: %s", namespace)
+		// ignore namespaces that are *NOT* prefixed with openshift- but accept metav1.NamespaceAll
+		if !(hasOpenshiftPrefix(namespace)) && namespace != metav1.NamespaceAll {
+			continue
+		}
+
+		nonCopiedCsvInformer := newNonCopiedCsvInformerForNamespace(namespace, config.ExternalClient(), config.ResyncPeriod()())
+
+		nonCopiedCsvQueue := workqueue.NewNamedRateLimitingQueue(
+			workqueue.DefaultControllerRateLimiter(),
+			fmt.Sprintf("%s/csv-ns-labeler-plugin-csv-queue", namespace),
```

Comment:
shouldn't this be plugin.log() or something?

Reply:
this isn't a log statement, it's the name of the queue - just using string formatting to build it